jung-algorithms/src/main/java/edu/uci/ics/jung/layout/model/LayoutModel.java

```diff
@@ -84,39 +109,44 @@
 
   /**
    * This exists so that LayoutModel will not have dependencies on java awt or swing event classes
-   *
-   * @return
+   * This event type tells the viewing system that it should re-draw itself to show the latest
```

Comment:
I'll document it in AbstractLayoutModel where they all live. There are actually 3 kinds:
ChangeSupport replaces a dependency on swing ChangeEvent. It means 'repaint yourself now'
It has no payload
LayoutChangeSupport  sends an event with a node and a (new) location
LayoutStateChangeSupport sends an event with a LayoutModel and a boolean. The boolean means "I am (or am not) busy moving things around now". It is important for the spatial data structures so they know not to do useless work competing with the relaxer thread.
This is also in the issue I submitted describing how things work. It will probably evolve and get better.

Events really complicate things, which is why i have a lot of functional programming style methods to pass what is needed.

Reply:
Thanks, I appreciate the explanation (and the plans to add more documentation).

Maybe we should consider renaming ChangeSupport to something like VisualizationUpdater?

Reply:
I have trouble keeping them all straight. Better names would help a lot, along with precise descriptions where they are declared in AbstractLayoutModel. 
I may be able to get rid of the first one, as i believe that any viz will want to repaint when any node changes location.
I think the real purpose of the time-to-repaint event is for the Viz model to tell the VizViewer to repaint.
I'll look into this for a future PR

Reply:
Don't take this the wrong way, but if _you_ have trouble keeping them all straight, I agree that that's a sign that we may want to rethink this part of the design.

Thanks for continuing to think about this.

Reply:
I'm renaming and reworking the event models. I would like it to be very clear who does what.